freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -181,4 +195,35 @@ private boolean isDuplicateEntryException(FreeIpaClientException e) {
                 .filter(c -> c == DUPLICATE_ENTRY_ERROR_CODE)
                 .isPresent();
     }
+
+    private String getDecodedKeytab(String encodeKeytab) {
+        byte[] decodedBytes = Base64.getDecoder().decode(encodeKeytab);
+        return new String(decodedBytes);
```

Comment:
I think converting the binary data into a `String` has unintended consequences that are based on the JVM configuration. Here is the documentation for the constructor: "Constructs a new String by decoding the specified array of bytes using the platform's default charset."

I know I said that we should probably put the binary data in the vault, but that was before I knew what our the API is for `SecretService`. It looks like that isn't a great idea based on the API which takes a string.

Reply:
the whole KerberosMgmtV1Service handles the keytab as String. should it be switched to sth else?

Reply:
if it has to be binary and we can't base64 encode it for some reason, it still could be encrypted and the key stored in vault. but in the first run I think we should try the base64 way.

Reply:
you are correct, it's not a good idea to store binary data in String. I will change it.

Base64 is a way to encode binary data into an ASCII character set known to pretty much every computer system, in order to transmit the data without loss or modification of the contents itself.
I think KerberosMgmtV1Service should put the base64 encoded string into the vault. It should be converted back to binary format just before it's written to a file. Bottom line, keytab should be in base64 encoded format when it is moved around.

If you have a different opinion, let me know.

Reply:
I meant the FreeIPAClient should return with byte array if it's not a string, and we should have only base64 encoded stuff in String then. This still should be fixed